### ISSUE TITLE
Removing Hardware Key protection (YubiKey) from database results in "invalid credential" error

### ISSUE THREAD

**Issue by goodc0re** [keepassxreboot/keepassxc#4041]
Removing Hardware Key protection (YubiKey) from a kdbx database results in "invalid credential" error when opening the database from that point onwards, after saving it. Database can then no longer be opened with or without yubikey present. 

This is the error: 

<img width="565" alt="image" src="https://user-images.githubusercontent.com/2564288/71160322-a38a6580-2247-11ea-81b6-aeab63a82e21.png">

## Expected Behavior
KeePassXC should open the database with only a password, after the Hardware key protection has been removed and the database was saved. It should open without a YubiKey present. 

## Current Behavior
The KeePassXC database can not be opened after entering just the password. No matter if the YubiKey that was used before removing the additional protection is selected and plugged in or not. The error shown above occurs in both cases. The database can no longer be opened. 

## Steps to Reproduce
Create a new KeePassXC database. 
Set a password and also a YubiKey for additional protection. 
Lock the database. 
Unlock with password and YubiKey. 
Go to Database, Change Master Key, and remove the Yubikey protection. 
Save and close. 
Open KeePassXC and open the database again, enter the password. 
The YubiKey is still being populated in the Hardware Key pulldown list. 
KeePass will still prompt to Touch the YubiKey, after touching the YubiKey the above error shows. 

When entering the password and instead setting the pulldown list to "Select ...", thereby deselecting the yubikey, the above error also occurs. 

Now the strangest thing is that when opening a previously saved backup file of the database, that was saved with YubiKey protection still in place, without ever having attempted to remove the Yubikey protection, it can not be opened either, with the YubiKey being detected, but after touching the YubiKey the same error above is shown. 

This also happens when closing KeePassXC inbetween each database opening and closing, no difference. 

## Context
Now, even with the Yubikey being detected, it is not possible to open any database. 

## Debug Info
KeePassXC - Version 2.5.1
Revision: 0fd8836

Qt 5.13.1
Debugging mode is disabled.

Operating system: macOS Mojave (10.14)
CPU architecture: x86_64
Kernel: darwin 18.7.0

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- TouchID

Cryptographic libraries:
 libgcrypt 1.8.5

Libraries:
- LIBS

Operating system: OS
CPU architecture: ARCH
Kernel: KERNEL

**Comment by phoerious** [comment]
Double check that you have all credentials entered that are needed and none else. In particular, make sure you do not enter your KDBX file in the key file field.

**Comment by goodc0re** [comment]
Yes of course, I have not only double-checked, I have done the entire process detailed under the Steps to Reproduce section with a completely new database and can reproduce it every time. 

If only the password is entered and the Hardware Key field is set to "Select.." the error occurs, no other credentials entered or selected, no key file selected.

**Comment by droidmonkey** [comment]
This has been brought up before and I couldn't replicate it. I will try your steps.

**Comment by droidmonkey** [comment]
Is your database kdbx 3.1 based??

**Comment by droidmonkey** [comment]
This is a legitimate bug and has been reproduced. The fix is going to be in 2.5.3. See #4147. I was unable to reproduce before because it only impacts KDBX 3.1.

**Comment by goodc0re** [comment]
> @goodc0re Is your database kdbx 3.1 based??

3.1 yes, sorry for the late reply
